File: restaurant/state/ordered_state.py
from state.dish_state import DishState

class OrderedState(DishState):
    """Trạng thái: Đơn hàng vừa được tạo"""
    
    def handle(self, context):
        """Xử lý khi đơn hàng được đặt"""
        print(f"Đơn hàng {context.order_id} vừa được tạo.")
        context.state = OrderedState()
    
    def get_status_name(self) -> str:
        return "Ordered"

File: backend/src/main/java/com/restaurant/pattern/state/OrderedState.java
package com.restaurant.pattern.state;

import com.restaurant.model.entity.OrderItem;
import com.restaurant.model.enums.DishStatus;

/**
 * Trạng thái: Món vừa được đặt
 * Trạng thái tiếp theo: COOKING
 */

public class OrderedState implements DishState {
    @Override
    public void nextState(OrderItem orderItem) {
        // Chuyển sang trạng thái COOKING
        orderItem.setDishStatus(DishStatus.COOKING);
        orderItem.setState(new CookingState());
        System.out.println("Món '" + orderItem.getMenuItem().getName()
                + "' đã chuyển sang trạng thái: COOKING");
    }

    @Override
    public String getStateName() {
        return "ORDERED";
    }
}